lib/new_relic/agent/instrumentation/rdkafka/prepend.rb

```diff
@@ -36,7 +36,8 @@ module RdkafkaConfig
     module Prepend
       include NewRelic::Agent::Instrumentation::RdkafkaConfig
 
-      if defined?(::Rdkafka) && Gem::Version.new(::Rdkafka::VERSION) >= Gem::Version.new('0.16.0')
+      if (defined?(::Rdkafka) && Gem::Version.new(::Rdkafka::VERSION) >= Gem::Version.new('0.16.0')) ||
+          (Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.7.0'))
```

Comment:
@tannalynn Should the `defined?(::Rdkafka)` check be applied no matter what? With this change, a Ruby version over 2.7.0 will cause the `defined?` check to be skipped.

Reply:
Do we need to check if Rdkafka is defined? 

I thought this file would only load if it's already defined because of the dependency detection check: https://github.com/newrelic/newrelic-ruby-agent/blob/b9c118e1c51b6940a0199416d17c6a9c5b1501bf/lib/new_relic/agent/instrumentation/rdkafka.rb#L5-L18 

If we need to check it, should the `chain` file also have a `defined?(::Rdkafka)` check?

Reply:
When the agent is running normally it's not a problem, this is something I did in order to fix a test that was failing related to loading instrumentation. https://github.com/newrelic/newrelic-ruby-agent/pull/2849
IIRC, that test will load every single file and make sure no errors are thrown when loading it, so of course this fails because it assumes rdkafka is present already. Adding the defined check was the simplest path forward with no real side effect since it was just an extra check one time on start up.

Also, It's not needed in chain because in the chain file, that code is in a method that gets called later, so it works differently than prepend.